From e072eb0ea192433a6ef68611dc5ca00924df6457 Mon Sep 17 00:00:00 2001
From: Slawomir Rosek <slawomir.rosek@semihalf.com>
Date: Mon, 5 Jun 2017 16:25:08 +0200
Subject: [PATCH 250/375] octeontx-pko: fix PTF_IOBP_CFG magic address

Any descriptor segment that does not have a native data read fetch
associated with it (eg. SEND_HDR+SEND_JUMP, SEND_HDR+SEND_IMM)
still needs to generate an NCB read to ensure data ordering.

Signed-off-by: Slawomir Rosek <slawomir.rosek@semihalf.com>
---
 drivers/net/ethernet/cavium/octeontx-83xx/pkopf_main.c | 18 +++++++++++++-----
 1 file changed, 13 insertions(+), 5 deletions(-)

diff --git a/drivers/net/ethernet/cavium/octeontx-83xx/pkopf_main.c b/drivers/net/ethernet/cavium/octeontx-83xx/pkopf_main.c
index c040e6f48f8c..211e13912250 100644
--- a/drivers/net/ethernet/cavium/octeontx-83xx/pkopf_main.c
+++ b/drivers/net/ethernet/cavium/octeontx-83xx/pkopf_main.c
@@ -812,12 +812,20 @@ static int pko_mac_init(struct pkopf *pko, int mac_num, int mac_mode)
 	reg = (rate << 3) | size;
 	pko_reg_write(pko, PKO_PF_PTGFX_CFG(ptgf), reg);
 
-	reg = (1ull << 63) | 0x10; /* 0x10 -- recommended in HRM.*/
-	/* Note: For XFI interface, this value may be big and can create
-	 * "underflow" condition in the BGX TX FIFO. If this happens,
-	 * use value = 3..6.
-	 */
+	reg = (1ull << 63)
+	       | fpa->pool_iova /* dummy read address -- required by any
+				 * descriptor segment that does not have
+				 * a native data read fetch associated
+				 * with it (eg. SEND_JUMP, SEND_IMM).
+				 */
+	       | 0x10; /* 0x10 -- recommended in HRM.*/
+	       /* Note: For XFI interface, this value may be big and can create
+		* "underflow" condition in the BGX TX FIFO. If this happens,
+		* use value = 3..6.
+		*/
+
 	pko_reg_write(pko, PKO_PF_PTF_IOBP_CFG, reg);
+
 	return 0;
 }
 
-- 
2.14.1

